spark-3.1-spanner-lib/src/main/java/com/google/cloud/spark/spanner/SpannerRelation.java

```diff
@@ -38,6 +38,19 @@ public SpannerRelation(String table, SparkSession session) {
     this.session = session;
   }
 
+  private String buildSQL(String[] requiredColumns, Filter[] filters) {
+    // 1. Create the conjuction of filters.
```

Comment:
Notice that when `df.count()` is called, `requiredColumns` is an empty array

Reply:
That code is no longer relevant as we are only supporting newer dialects. Thanks for the review though David!